hmt_escrow_js/Job.js

```diff
@@ -0,0 +1,97 @@
+require('dotenv').config()
+
+const keythereum = require("keythereum")
+const ETHInterface = require('./ETHInterface')
+const Ajv = require('ajv')
+const manifestSchema = require('./manifest-schema-pydantic')
+const request = require('request');
+
+class Job {
+  constructor(gas_payer, gas_payer_priv, rep_oracle_pub_key, manifest_url) {
+    this._gas_payer = gas_payer
+    this._gas_payer_priv = gas_payer_priv
+    this._rep_oracle_pub_key = rep_oracle_pub_key
+    this._manifest_url = manifest_url
+
+    this.serialized_manifest = this._download_manifest()
+    this.amount = null
+
+    // Validate manifest
+    this._validate_manifest()
+
+    if(!this._validate_credentials()) {
+      throw Error('Invalid Combination of Public & Private key')
+    }
+
+    // Process Manifest
+    this._process_manifest()
+  }
+
+  _download_manifest() {
+    // Download manifest if URL
+    // Support for future stuff too like s3, direct upload etc.
+    request(this._manifest_url, function (error, response, body) {
+      if (!error && response.statusCode == 200) {
+        return JSON.parse(body);
+      }
+      else {
+        throw new Error("Couldn't download manifest file, please try again.")
+      }
+    })
+  }
+
+
+  _validate_credentials() {
+    const calculated_pub_key = ETHInterface.web3Instance.utils.toChecksumAddress(keythereum.privateKeyToAddress(this._gas_payer_priv))
+    return ETHInterface.web3Instance.utils.toChecksumAddress(this._gas_payer) === calculated_pub_key
+  }
+
+  _validate_manifest() {
+    const ajv = new Ajv()
+    const validate = ajv.compile(manifestSchema)
+    if(!validate(this.serialized_manifest)) {
+      throw new Error("Manifest invalid")
+    }
+  }
+
+  _process_manifest() {
+    this.amount = Math.round(this.serialized_manifest.task_bid_price * this.serialized_manifest.job_total_tasks)
+  }
+
+  
+  async launch() {
+    // Only handling launch case where fresh factory is created for each escrow, in future, expand this to mimic python lib
```

Comment:
Oh, snap! You gotta launch a factory only and only once per hmtoken! And that factory handles all escrows. Please "mimic" this python lib functionality before the merging of this PR.